core/store/src/trie/insert_delete.rs

```diff
@@ -432,86 +435,84 @@ impl Trie {
                 }
             }
         }
-        self.fix_nodes(memory, path, key_deleted)?;
+
+        let mut child_memory_usage = 0;
+        for handle in path.into_iter().rev() {
+            // First, recompute memory usage, emulating the recursive descent.
+            let TrieNodeWithSize { node, mut memory_usage } = memory.destroy(handle);
```

Comment:
I do not have much context, but is it ok that we do `memory.destroy(handle)` both here and inside `squash_node()` ?

Reply:
Haha! Yeah, looking at the code, that's what I thought as well, but seems like we are calling `memory.store_at` right after calling destroy which undoes this.